here-naksha-lib-model/src/commonMain/kotlin/naksha/model/request/PropertyFilter.kt

```diff
@@ -0,0 +1,89 @@
+package naksha.model.request
+
+import naksha.base.AnyObject
+import naksha.base.Platform
+import naksha.jbon.JbFeatureDecoder
+import naksha.model.request.query.*
+
+class PropertyFilter(val req: ReadFeatures) : ResultFilter {
+
+    companion object {
+        const val PROPERTIES = "properties"
+    }
+
+    override fun call(a1: ResultTuple): ResultTuple? {
+        val pSearch = req.query.properties ?: return a1
+        if (a1.tuple == null) return null
+        if (a1.tuple!!.feature == null) return null
+        val decoder = JbFeatureDecoder()
+        decoder.mapBytes(a1.tuple!!.feature!!)
+        if (resolvePropsQuery(pSearch, decoder)) return a1
+        return null
+    }
+
+    private fun resolvePropsQuery(pQuery: IPropertyQuery?, decoder: JbFeatureDecoder) : Boolean {
+        when (pQuery) {
+            null -> return true
+            is PAnd -> {
+                pQuery.forEach {
```

Comment:
I think you can simplify it with:
`return pQuery.all { resolvePropsQuery(it, decoder) }`

and below:
`return pQuery.any { resolvePropsQuery(it, decoder) }`